core/src/test/scala/org/apache/spark/scheduler/CoarseGrainedSchedulerBackendSuite.scala

```diff
@@ -179,6 +179,53 @@ class CoarseGrainedSchedulerBackendSuite extends SparkFunSuite with LocalSparkCo
     }
   }
 
+  // Every item corresponds to (CPU resources per task, GPU resources per task,
+  // and the GPU addresses assigned to all tasks).
+  Seq(
+    (1, 1, Array(Array("0"), Array("1"), Array("2"), Array("3"))),
+    (1, 2, Array(Array("0", "1"), Array("2", "3"))),
+    (1, 4, Array(Array("0", "1", "2", "3"))),
+    (2, 1, Array(Array("0"), Array("1"))),
+    (4, 1, Array(Array("0"))),
+    (4, 2, Array(Array("0", "1"))),
+    (2, 2, Array(Array("0", "1"), Array("2", "3"))),
```

Comment:
does it generate deterministic result `Array(Array("0", "1"), Array("2", "3")))` ?

I think it might also be Array(Array("0", "2"), Array("1", "3"))) or `Array(Array("1", "0"), Array("3", "2")))`

Reply:
Yeah, that's true, it will generate the deterministic result. The internal implementation will offer the sorted GPU addresses to the tasks (to align with the old version)